spec/models/game_spec.rb

```diff
@@ -148,11 +148,46 @@
     end
 
     it 'returns false if king is currently in check' do
-      game = FactoryGirl.create(:game)
+      game = FactoryGirl.create(:game, move_number: 10)
       white_king = FactoryGirl.create(:king, is_white: true, game: game, x_position: 4, y_position: 7)
-      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 4, y_position: 5)
+      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 4, y_position: 5, game_move_number: 10)
       FactoryGirl.create(:king, is_white: false, game: game, x_position: 5, y_position: 2)
       expect(game.stalemate?(white_king.is_white)).to eq(false)
     end
   end
+
+  describe '#checkmate?' do
+    it 'should return true if king is in checkmate' do
+      game = FactoryGirl.create(:game, move_number: 15)
+      FactoryGirl.create(:king, is_white: false, game: game, x_position: 8, y_position: 5)
+      FactoryGirl.create(:king, is_white: true, game: game, x_position: 6, y_position: 5)
+      FactoryGirl.create(:rook, is_white: true, game: game, x_position: 8, y_position: 1, game_move_number: 15)
+      expect(game.checkmate?).to eq(true)
+    end
+
+    it 'should return false if king can capture attacking piece' do
+      game = FactoryGirl.create(:game)
+      FactoryGirl.create(:king, is_white: true, game: game, x_position: 6, y_position: 3)
+      FactoryGirl.create(:king, is_white: false, game: game, x_position: 5, y_position: 8)
+      FactoryGirl.create(:rook, is_white: false, game: game, x_position: 5, y_position: 3)
+      expect(game.checkmate?).to eq(false)
+    end
+
+    it 'should return false if friendly piece can block checkmate' do
+      game = FactoryGirl.create(:game)
+      FactoryGirl.create(:king, is_white: false, game: game, x_position: 3, y_position: 8)
+      FactoryGirl.create(:king, is_white: true, game: game, x_position: 5, y_position: 1)
+      FactoryGirl.create(:bishop, is_white: false, game: game, x_position: 4, y_position: 7)
+      FactoryGirl.create(:bishop, is_white: true, game: game, x_position: 6, y_position: 5)
+      expect(game.checkmate?).to eq(false)
+    end
```

Comment:
I don't see how this test is supposed to check if friendly piece can stop a checkmate. In this scenario the king (I think it is black king that is supposed to be threatened) is not even in check. I think what you should actually go for is situation where black bishop is located at (3, 6), and black king is held in check by white bishop but black player can move black bishop between black king and white bishop. Even better would be situation where black king is in check and cannot move at all, and only option for black player is to move bishop between black king and the attacking piece.

Reply:
I adjusted this so the black king is in check and cannot move at all, per your suggestion.

Reply:
The same pieces are in the same positions as they were before, black king is currently not in check. You can see this if you write out the names of the pieces in an excel sheet.

Reply:
I think the black king is in check. The white bishop at (6, 5) is putting the king in check. And the black bishop at (3, 6) can move to (4, 7) to block the check.